src/main/java/com/teragrep/lsh_01/Main.java

```diff
@@ -60,18 +61,22 @@ public static void main(String[] args) {
         RelpConnectionFactory relpConnectionFactory = new RelpConnectionFactory(relpConfig);
         Pool<IManagedRelpConnection> pool = new Pool<>(relpConnectionFactory, new ManagedRelpConnectionStub());
 
-        RelpConversion relpConversion = new RelpConversion(
-                pool,
-                securityConfig,
-                basicAuthentication,
-                lookupConfig,
-                payloadConfig
-        );
+        IMessageHandler conversion = new RelpConversion(pool, securityConfig, basicAuthentication, lookupConfig);
+
+        // apply splitting if configured
+        switch (payloadConfig.splitType) { // payloadConfig is Validateable, no need to check wrong config here
```

Comment:
could switch-case be replaced with another object?

Reply:
Tried to refactor code for creating a decorator for this in NettyHttpServer and even some other objects that use / forward the RelpConversion object. This was so that I could put the config checking code there so it doesn't clutter the other code. This however seems to be impossible now with objects extending and implementing 3rd party objects that are not immutable. Would need a complete refactoring really.

Would it be terrible to create an object just for creating this Conversion object with the config? Not really OOP style, but the code would be somewhere else than in the Main class.

Reply:
Made a new object called ConversionFactory that provides the correctly configured object. However, the switch case remains which I think is inevitable with a String config option.